### PR TITLE
Add LocationApi

### PR DESCRIPTION
**Pull request by jtjk**

### CODE REVIEW

File: azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/features/LocationApi.java [hunk 1]
```diff
@@ -38,9 +41,12 @@
  *
  * @see <a href="http://msdn.microsoft.com/en-us/library/gg441299">docs</a>
  */
-@Path("/locations")
+@Path("/subscriptions/{subscriptionid}")
 @Headers(keys = "x-ms-version", values = "{jclouds.api-version}")
```
Comment (jmspring) [inline azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/features/LocationApi.java:45]: The x-ms-version header isn't needed for ARM.  Harmless, however.

**Comment by jmspring** [comment]
See above note re: x-ms-version.

Note, that in the tests the "version" for the ARM REST API is hard coded in the API tests.  Should this be a contact somewhere?

Otherwise LGTM

**Comment by ritazh** [comment]
For documentation purpose, to run LocationApiLiveTest:

```
mvn -Dtest=LocationApiLiveTest -Dtest.azurecompute-arm.identity="" -Dtest.azurecompute-arm.subscriptionid="" -Dtest.azurecompute-arm.credential="" -Dtest.azurecompute-arm.endpoint="https://management.azure.com" -Dtest.jclouds.oauth.resource="https://management.azure.com/" -Dtest.oauth.endpoint="https://login.microsoftonline.com/[TenantId]/oauth2/token" test
```

File: azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/compute/AzureComputeServiceAdapter.java [hunk 2]
```diff
@@ -260,9 +260,12 @@ public boolean apply(final OSImage input) {
                               image.eula());
    }
 
+   private String getSubscriptionId() {
```
Comment (ritazh) [inline azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/compute/AzureComputeServiceAdapter.java:263]:
```
 protected String getSubscriptionId() {
      String subscriptionId = null;
      if (System.getProperties().containsKey("test.azurecompute-arm.subscriptionid"))
         subscriptionId = System.getProperty("test.azurecompute-arm.subscriptionid");
      assertNotNull(subscriptionId);
      return subscriptionId;
   }
```

File: azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/AzureComputeApi.java [hunk 3]
```diff
@@ -67,7 +67,7 @@
     * @see <a href="http://msdn.microsoft.com/en-us/library/gg441299">docs</a>
```
Comment (ritazh) [inline azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/AzureComputeApi.java:67]: https://msdn.microsoft.com/en-US/library/azure/dn790540.aspx

File: azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/features/LocationApi.java [hunk 4]
```diff
@@ -38,9 +41,12 @@
  *
  * @see <a href="http://msdn.microsoft.com/en-us/library/gg441299">docs</a>
```
Comment (ritazh) [inline azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/features/LocationApi.java:42]: https://msdn.microsoft.com/en-US/library/azure/dn790540.aspx

File: azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/features/LocationApi.java [hunk 5]
```diff
@@ -22,14 +22,17 @@
...
+import org.jclouds.azurecomputearm.oauth.v2.filters.OAuthFilter;
 import org.jclouds.rest.annotations.Headers;
-import org.jclouds.rest.annotations.XMLResponseParser;
+import org.jclouds.rest.annotations.QueryParams;
+import org.jclouds.rest.annotations.RequestFilters;
+import org.jclouds.rest.annotations.SelectJson;
+import org.jclouds.rest.annotations.Fallback;
 
 /**
  * The Service Management API includes operations for listing the available data center locations for a cloud service in
```
Comment (ritazh) [inline azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/features/LocationApi.java:38]: Update to reference Azure Resource Manager API

File: azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/AzureComputeApi.java [hunk 6]
```diff
@@ -67,7 +67,7 @@
...
     */
    @Delegate
-   LocationApi getLocationApi();
+   LocationApi getLocationApi(@PathParam("subscriptionid") String subscriptionid);
 
    /**
     * The Service Management API includes operations for managing the cloud services beneath your subscription.
```
Comment (ritazh) [inline azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/AzureComputeApi.java:73]: Update to reference "Azure Resource Manager API"

File: azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/compute/AzureComputeServiceAdapter.java (hunk 2 above)
Comment (jtjk) [inline azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/compute/AzureComputeServiceAdapter.java:263]: We should not try to access test parameters from non-test code. It is better to have todo there as that class needs complete rewrite anyway.

**Comment by jtjk** [comment]
Updated based on comments. Obsolete version header removed. Document links fixed. AzureComputeServiceAdapter was not changed as we should not add test code there and we have own trello-item for that work as it need complete rewrite.

**Comment by ritazh** [comment]
LGTM. Please rebase against latest on azurecompute-arm so we can merge it.

**Pull request merged**